### PR TITLE
[NC-2005] Implemented --p2p-enabled configuration item

### PR DESCRIPTION
**Pull request by mark-terry**
## PR description
Implemented a `--p2p-enabled` configuration item, that enables the complete disabling of the p2p functionality within Pantheon.

## Fixed Issue(s)

#NC-2005

### CODE REVIEW

File: pantheon/src/test/java/tech/pegasys/pantheon/cli/PantheonCommandTest.java [hunk 1]
```diff
@@ -528,6 +529,35 @@ public void genesisPathDisabledUnderDocker() {
     assertThat(commandOutput.toString()).isEmpty();
   }
 
+  @Test
```
Comment (lucassaldanha) [inline pantheon/src/test/java/tech/pegasys/pantheon/cli/PantheonCommandTest.java]: Do we need a test to verify that the default behaviour when `--p2p-enabled` isn't used is p2p-enabled = true?

File: pantheon/src/test/java/tech/pegasys/pantheon/cli/PantheonCommandTest.java [hunk 2]
```diff
@@ -528,6 +529,35 @@ public void genesisPathDisabledUnderDocker() {
...
+    parseCommand("--p2p-enabled");
+
+    verify(mockRunnerBuilder.p2pEnabled(eq(true))).build();
+
+    assertThat(commandOutput.toString()).isEmpty();
+    assertThat(commandErrorOutput.toString()).isEmpty();
+  }
+
+  @Test
+  public void enableP2pOptionMustBeUsed() {
```
Comment (lucassaldanha) [inline pantheon/src/test/java/tech/pegasys/pantheon/cli/PantheonCommandTest.java]: The name of this test has nothing to do with what is being tested, has it?

Comment (mark-terry) [reply pantheon/src/test/java/tech/pegasys/pantheon/cli/PantheonCommandTest.java]: I'll rework that a bit...

File: pantheon/src/test/java/tech/pegasys/pantheon/cli/PantheonCommandTest.java (hunk 1 above)
Comment (mark-terry) [reply pantheon/src/test/java/tech/pegasys/pantheon/cli/PantheonCommandTest.java]: Being implicitly tested now as part of the Runner tests.

**Pull request merged**